packages/extension-ui/src/Popup/Accounts/EditAccountMenu.tsx

```diff
@@ -3,41 +3,177 @@
+  const { show } = useToast();
 
-  return <>
-          <Header
-            showBackArrow
-            showHelp
-            text={t<string>('Edit Account')}
-        />
-          {/* TODO: */}
-        <div className={className}>PLACEHOLDER</div>
+  function findAccountInHierarchy(accounts: AccountJson[], _address: string) {
+    return hierarchy.find(({ address }): boolean => address === _address) || null;
+  }
+
+  const _onCopy = useCallback(() => show(t<string>('Public address copied to your clipboard'), 'success'), [show, t]);
+
+  const [account, setAccount] = useState(findAccountInHierarchy(accounts, address));
+
+  const [isHidden, setIsHidden] = useState(account?.isHidden);
+
+  console.log('account', account);
+
+  const chain = useMetadata(account?.genesisHash, true);
+
+  const settings = useContext(SettingsContext);
+
+  const prefix = chain ? chain.ss58Format : settings.prefix === -1 ? 42 : settings.prefix;
+
+  const theme = (account && account.type === 'ethereum' ? 'ethereum' : chain?.icon || 'polkadot') as IconTheme;
 
-  </>;
+  const _toggleVisibility = useCallback((): void => {
+    console.log('toggleVisibility from address', address, isHidden);
+    address &&
+      showAccount(address, isHidden || false)
+        .then((data) => {
+          if (account) {
+            setAccount({
+              ...account,
+              isHidden: !account.isHidden
+            });
+          }
+
+          setIsHidden(data);
+        })
+        .catch(console.error);
+  }, [address, isHidden, account]);
+
+  return (
+    <>
+      <Header
+        showBackArrow
+        showHelp
+        text={t<string>('Edit Account')}
+      />
+      <div className={className}>
+        <Identicon
+          className='identityIcon'
+          iconTheme={theme}
+          isExternal={isExternal === 'true'}
+          onCopy={_onCopy}
+          prefix={prefix}
+          value={address}
+        />
+        <EditMenuCard
+          description={account?.name || ''}
+          extra='chevron'
+          position='top'
+          title='Name'
+        />
+        <CopyToClipboard text={(address && address) || ''}>
+          <EditMenuCard
+            description={ellipsisName(address) || ''}
+            extra='copy'
+            onClick={_onCopy}
+            position='middle'
+            title='Address'
+          />
+        </CopyToClipboard>
+        <EditMenuCard
+          description={chain?.genesisHash ? 'testnet' : 'Mainnet'}
+          extra='chevron'
+          position='middle'
+          title='Network'
+        />
+        <EditMenuCard
+          description=''
+          position='bottom'
+          title='Visibility for apps'
+          toggle={
+            <>
+              <Switch
+                checked={!account?.isHidden || false}
+                checkedLabel={t<string>('')}
+                onChange={_toggleVisibility}
+                uncheckedLabel={t<string>('')}
```

Comment:
Does the fact we have empty strings here are correct? Why do we translate empty string? If you need labels empty we could just put null or empty string altogether. And if this is something that will be added later then a `todo` would be nice :)

Reply:
good point 😅